### DISCUSSION TITLE
Blank map when moved to production for android

### DISCUSSION THREAD

**Discussion by jhexdejesus1** [react-native-maps/react-native-maps#4311] [Q&A] [answered]
Hi,

Google maps is not showing properly when app is moved to production. My project is made in expo, it was working fine on expo tests. But when I made an apk build, maps is blank. On iOs is working fine.

![image](https://user-images.githubusercontent.com/57471351/162670192-aac41d87-bcc2-4adf-86c2-860042cd06ec.png)

**Comment by jhexdejesus1** [comment]
Versions:
"expo": "~43.0.2",
"react-native-maps": "^0.30.1",

Source Code:
<React.Fragment>
            <MapView
              ref={(map) => {
                this.map = map;
              }}
              data={markers}
              initialRegion={initialRegion}
              renderMarker={renderMarker}
              showsUserLocation={true}
              showsMyLocationButton={true}
              onUserLocationChange={(event) =>
                this.setState({ userCoordinate: event.nativeEvent.coordinate })
              }
              style={[
                StyleSheet.absoluteFill,
                { marginBottom: this.state.marginBottom, },
              ]}
              textStyle={{ color: "#bc8b00" }}
              containerStyle={{
                backgroundColor: "#fff",
                borderColor: "#bc8b00",
              }}
              onMapReady={() => {
                this.setState({ marginBottom: 0 });
              }}
              mapPadding={{ top: 100 }}
            >
              {this.state.markers.map((marker) => (
                <React.Fragment key={marker.identifier}>
                  <Marker
                    key={marker.identifier}
                    coordinate={{
                      latitude: marker.latitude,
                      longitude: marker.longitude,
                    }}
                    title={marker.identifier}
                    description={marker.identifier}
                  />
                </React.Fragment>
              ))}
              {this.state.markers.map((marker) => (
                <React.Fragment key={marker.identifier}>
                  <MapView.Circle
                    key={marker.identifier}
                    center={{
                      latitude: marker.latitude,
                      longitude: marker.longitude,
                    }}
                    radius={marker.radius}
                    strokeColor={styles.circleStrokeColor.color}
                    fillColor={styles.circleFillColor.color}
                    strokeWidth={2}
                  />
                </React.Fragment>
              ))}
            </MapView>
          </React.Fragment>

**Comment by princefishthrower** [comment]
We're seeing the same... specifically our map `Marker` is not displaying on real Android Devices (works everywhere else, iOS, and Android emulator)

**Comment by jhexdejesus1** [comment]
I think this happened when they added 0.31.0-beta.2. My app is working fine in previous months, then last 2 weeks ago it went blank.

**Comment by princefishthrower** [comment]
Ah ok, likely unrelated to our problem then, we are on 0.30.1

**Comment by adamsoutar** [comment]
@jhexdejesus1 Maybe check on your environment variables?
I'm not certain how it works with Expo, but I had a "blank map" issue that ended up being my build process unable to find the `GOOGLE_MAPS_API_KEY`

**Comment by russmenum** [comment]
Unrelated to your issue @jhexdejesus1 , but since you have this working in iOS, did you have a build error when you first started?  what I am getting [here](https://github.com/react-native-maps/react-native-maps/issues/4183)

My app is an EXPO ejected app so thought you might have? If so, what did you do to fix it?

I had this error with the blank screen on Android, if you do a release build in Android Studio instead of dev for the sim are you getting a red error about not finding the key? And if so did you try doing the create google_maps_api.xml in android/app/src/main/res/values work round?

**Comment by russmenum** [comment]
@princefishthrower same question since you also have this working in iOS. If you had the 

```
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

```

error what is the current fix?

**Comment by jhexdejesus1** [comment]
> Unrelated to your issue @jhexdejesus1 , but since you have this working in iOS, did you have a build error when you first started? what I am getting [here](https://github.com/react-native-maps/react-native-maps/issues/4183)
> 
> My app is an EXPO ejected app so thought you might have? If so, what did you do to fix it?
> 
> I had this error with the blank screen on Android, if you do a release build in Android Studio instead of dev for the sim are you getting a red error about not finding the key? And if so did you try doing the create google_maps_api.xml in android/app/src/main/res/values work round?

@russmenum react-native-maps is working fine ever since I made the app and haven't made any changes in my code. Then 2 weeks ago the issue for Android happened.

**Comment by russmenum** [comment]
> @russmenum react-native-maps is working fine ever since I made the app and haven't made any changes in my code. Then 2 weeks ago the issue for Android happened.


@jhexdejesus1 for my sanity can you glance at the ```Podfile``` and ```AppDelegate.m``` from my ticket and see if I added the code same place you did? At this point, I am just stuck. It is very odd to just have a build error like this for no reason.

**Comment by jhexdejesus1** [comment]
> > @russmenum react-native-maps is working fine ever since I made the app and haven't made any changes in my code. Then 2 weeks ago the issue for Android happened.
> 
> @jhexdejesus1 for my sanity can you glance at the `Podfile` and `AppDelegate.m` from my ticket and see if I added the code same place you did? At this point, I am just stuck. It is very odd to just have a build error like this for no reason.

@russmenum sorry, I can't help you with that I'm still new to react native and currently using Expo for my projects, it doesn't have ios nor android folder.

**Comment by chaban-sergey** [comment]
Has anyone been able to solve this problem? The problem only occurs when building the apk. When connected to the project via USB, the map is displayed correctly on the phone.

**Comment by danieldanielecki** [comment]
I've had the same problem, and [enabling Google Maps API's](https://github.com/react-native-maps/react-native-maps/blob/master/docs/installation.md#the-map-background-is-blank-google-maps) in GCP helped.

**Comment by realjjaveweb** [comment]
@danieldanielecki strange, are you using expo? if so, what is your app.json config?
I have the API enabled, "Maps SDK for Android", Application restrictions set to "Android apps", API restrictions set to "Restrict key" and 1 API selected ("Maps SDK for Android").
Also, did you set up billing?
Because I did all of this, have the correct api key in app.json and the issue remains.

**Reply by danieldanielecki** [reply]
Yes, I'm using Expo: https://github.com/tamotam-com/tamotam-app

**Comment by njoromyke** [comment]
Facing same problem in production too. Can someone explain how to set the production key in expo app.json

**Comment by chaban-sergey** [comment]
The problem was solved by recreating the API key

**Comment by VishalMishra0** [comment]
facing the same problem in production, it worked perfectly in development phase without any error showing on debugging but while build .apk it will not render map screen

**Comment by siblonet** [comment]
Same thing here! Doesn’t anyone knows how to solve it?

**Comment by mihneamanolache** [comment]
I solved it by creating a new Google Maps API Key for the project and updated it in app.json:
```json
"ios": {
      "config": {
        "googleMapsApiKey": <YOUR_API_KEY>
      },
    },
"android": {
      "config": {
        "googleMaps": {
          "apiKey": <YOUR_API_KEY>
        }
      }
    },
```

**Reply by realjjaveweb** [reply]
Strange, did you set up billing or not?

**Reply by mihneamanolache** [reply]
I have set up billing for the new project.

**Comment by jhexdejesus1** [comment]
I'm still facing this issue. I added both the application SHA-1 key and the google play app signing key to the API. When I install the app through APK, the map shows. But, when I install the app through Google Play the map is blank.

**Reply by Santa0727** [reply]
Hello @jhexdejesus1 
I also have same problem.
When I build to apk through eas build command and install on my Android, the map is working well.
But, when I build to aab and then upload to google open testing, then it is not working.
Did you resolve it?

**Reply by Zaeem6100** [reply]
@Santa0727  How did you resolve this error

**Reply by Santa0727** [reply]
@Zaeem6100  
I've updated all package versions to latest and built again.
After that, it worked.

**Reply by Zaeem6100** [reply]
For me it's not working

**Comment by siblonet** [comment]
thanks helped

**Comment by russmenum** [comment]
You can also use ```react-native-config``` and ```.env``` if you prefer

**Comment by rayat927** [comment]
Facing the problem as well, but i am facing it in test mode as well, don't know where i am going wrong, I created the api key and used it in app.json but still it is showing blank map with google logo. here is my code

```
import MapView, {PROVIDER_GOOGLE,} from 'react-native-maps'

<MapView
            style={{flex: 1}}
            provider={PROVIDER_GOOGLE}
            mapType='mutedStandard'
            initialRegion={{
            latitude: 37.78825,
            longitude: -122.4324,
            latitudeDelta: 0.005, 
            longitudeDelta: 0.005,
            }}
        />
```

here is my app.json
```
"android": {
      "adaptiveIcon": {
        "foregroundImage": "./assets/adaptive-icon.png",
        "backgroundColor": "#FFFFFF"
      },
      "config": {
        "googleMaps": {
          "apiKey": <Api key>
        }
      }
    },
```

**Reply by MochiTarts** [reply]
I'm facing the same problem too. My app.json also looks like the above. Here are the dependencies in my package.json

```
  "dependencies": {
    "expo": "~46.0.6",
    "expo-status-bar": "~1.4.0",
    "react": "18.0.0",
    "react-native": "0.69.4",
    "react-native-maps": "0.31.1"
  },
```

**Comment by jhexdejesus1** [comment]
I'm planning to try and upgrade my Api Signing Key. Once I confirmed that it won't affect users who's still using old android phones, I will proceed to upgrade the Key. I will update you guys if this will work.

**Reply by Santa0727** [reply]
Hello @jhexdejesus1 
After you upgrade, did it work?

**Comment by Santa0727** [comment]
Hello everyone,
I also have the same problem.
When I build to apk through eas build command and install it on my Android, the map is working well.
But, when I build to aab and then upload to google open testing, then it is not working.

I updated Android SDK Google Map API Key again and then used it.
But, still the same problem.

Some one can help me, please?

**Reply by realjjaveweb** [reply]
Do you have billing set up? (I feel that may be the missing piece, it's your decision, but if you would - please let us know if it worked or not)

**Comment by Kristinn-Thor** [comment, marked as answer]
I encountered this issue but managed to resolve it. In my case, the problem stemmed from not including the Play Store app signing key certificate (SHA-1 fingerprint) in the Android restrictions section for the API key I was using. Initially, I believed that providing the release certificate fingerprint would suffice.
Hope this helps somebody 🤞🏼

**Reply by AizenSousuke** [reply]
is it the .pem or the .jks sha-1?

**Reply by Zaeem6100** [reply]
In my Case, I have added both, but it's not working. Any other Solution
Tech Stack:  Expo, Bun  

```
"expo": "^52.0.35", 
"react-native": "0.76.7",
"react-native-maps": "1.18.0",
```

**Reply by mirGmail** [reply]
Same issue and it's very frustrating!  I added both eas SHA-1 fingerprint and the Play Store SHA1 fingerprint, but still blank map.  Works fine in Expo Go.

**Reply by Zaeem6100** [reply]
How are you giving the APi keys,

**Reply by johannes-for** [reply]
> Same issue and it's very frustrating! I added both eas SHA-1 fingerprint and the Play Store SHA1 fingerprint, but still blank map. Works fine in Expo Go.

Did you solve it?

**Reply by Zaeem6100** [reply]
I get notice when you have pre build app (Android) it already have your dev key in it and when you build with expo go it didn't change. Maybe in production you need to add that key in manifest file. I resolved this issue but adding that production env key in manifest file during the production push.
